### PR TITLE
docs: update docs for the AWS X-Ray tracer

### PR DESCRIPTION
**Pull request by suniltheta**
Signed-off-by: Sunil Narasimhamurthy <sunnrs@amazon.com>

Commit Message: update envoy docs to contain information related to AWS X-Ray extension and update codeowners for the same extension
Additional Description:
Risk Level: Low
Testing: N/A
Docs Changes: The PR updates the envoy documentation to reference X-Ray as a tracing extension that is supported by envoy and also gives reference towards the specific http header that is provided by this extension.
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: source/extensions/tracers/xray/BUILD [hunk 1]
```diff
@@ -59,7 +59,6 @@ envoy_cc_extension(
     hdrs = ["config.h"],
     category = "envoy.tracers",
     security_posture = "robust_to_untrusted_downstream",
-    status = "wip",
```
Comment (htuch) [inline source/extensions/tracers/xray/BUILD:62]: I'm not sure we can bump the AWS extensions from `wip` or `alpha` and claim security until they move away from libcurl. https://github.com/envoyproxy/envoy/issues/11816. We have a very high CVE rate in this library, probably an order-of-magnitude higher than any other Envoy dep.

Comment (suniltheta) [reply source/extensions/tracers/xray/BUILD:62]: Hi @htuch, thank you for taking a look at this PR and for the comment. This doc specifically remove the "wip" for the AWS X-Ray tracing extension [API](https://storage.googleapis.com/envoy-pr/276708a/docs/api-v3/config/trace/v3/xray.proto.html) which doesn't use the libcurl module or curl utility. There are other AWS extension that use it as per mentioned here at [external_deps](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/external_deps.html?highlight=curl). And the AWS X-Ray doesn't depend on these other AWS extensions as well.

Comment (suniltheta) [reply source/extensions/tracers/xray/BUILD:62]: Referring to the [comment](https://github.com/envoyproxy/envoy/pull/8906#issuecomment-552538354) left by the original author of this AWS X-Ray extension on an old #8906 PR.

Comment (mattklein123) [reply source/extensions/tracers/xray/BUILD:62]: I concur that this extension doesn't use curl. @htuch are ok with this? Seems fine to no longer be WIP.

Comment (htuch) [reply source/extensions/tracers/xray/BUILD:62]: Objection overruled  :)

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Comment by suniltheta** [comment]
Thanks mattklein123 & htuch for your time.

**Pull request merged**